lib/BackgroundJobs/StorageCrawlJob.php

```diff
@@ -113,10 +114,10 @@ protected function run($argument): void {
 		}
 
 		try {
-			$ignoreAllFileids = $this->getIgnoreFileids(Constants::IGNORE_MARKERS_ALL);
-			$ignoreImageFileids = $this->getIgnoreFileids(Constants::IGNORE_MARKERS_IMAGE);
-			$ignoreVideoFileids = $this->getIgnoreFileids(Constants::IGNORE_MARKERS_VIDEO);
-			$ignoreAudioFileids = $this->getIgnoreFileids(Constants::IGNORE_MARKERS_AUDIO);
+			$ignoreAllFileids = $this->getIgnoreFileids($storageId, Constants::IGNORE_MARKERS_ALL);
+			$ignoreImageFileids = $this->getIgnoreFileids($storageId, Constants::IGNORE_MARKERS_IMAGE);
+			$ignoreVideoFileids = $this->getIgnoreFileids($storageId, Constants::IGNORE_MARKERS_VIDEO);
+			$ignoreAudioFileids = $this->getIgnoreFileids($storageId, Constants::IGNORE_MARKERS_AUDIO);
```

Comment:
Just from reading the code this could still fail if there are a lot of fileIds returned for that storage, right? 

So it might be needed to chunk up the queries further down to run them individually. In terms of potential oracle support at some point, we might even want to use [a chunk limit below 1000.](https://github.com/nextcloud/server/blob/master/lib/private/DB/QueryBuilder/QueryBuilder.php#LL264)

Reply:
The problem is that we use a notIn. So, we can't really chunk them up, can we?

Reply:
This is where we apply the 999 items limit: https://github.com/nextcloud/recognize/pull/493/files#diff-0e0812060d3e67d725c0c0f4c82dc1fa4e54ca12aff169737a78eb250efc2846R133

Reply:
Right, ok. The number of arguments is basically the same as parent folders that contain a `.ignore` files right? Then I think that should be fine for now. If that becomes an issue the only other way I'd see would be to move the check if a file should be ignored to the actual Classifier, but that also would come with the cost of individual queries for each file.